src/main/java/org/broadinstitute/hellbender/utils/codecs/GENCODE/GencodeGtfExonFeature.java

```diff
@@ -0,0 +1,145 @@
+                                           String geneName,
+                                           GeneTranscriptType transcriptType,
+                                           GeneTranscriptStatus transcriptStatus,
+                                           String transcriptName,
+                                           int exonNumber,
+                                           String exonId,
+                                           LocusLevel locusLevel,
+                                           ArrayList<OptionalField<?>> optionalFields,
+                                           String anonymousOptionalFields) {
+
+        return new GencodeGtfExonFeature(featureOrderNumber, chromosomeName, annotationSource, featureType, genomicStartLocation, genomicEndLocation, genomicStrand, genomicPhase, geneId, transcriptId, geneType, geneStatus, geneName, transcriptType, transcriptStatus, transcriptName, exonNumber, exonId, locusLevel, optionalFields, anonymousOptionalFields);
+    }
+
+    // ============================================================================================================
+
+
+    GencodeGtfCDSFeature                        cds = null;
+    GencodeGtfStartCodonFeature                 startCodon = null;
+    GencodeGtfStopCodonFeature                  stopCodon = null;
+
+    // ============================================================================================================
+
+    public GencodeGtfCDSFeature getCds() {
+        return cds;
+    }
+
+    public GencodeGtfStartCodonFeature getStartCodon() {
+        return startCodon;
+    }
+
+    public GencodeGtfStopCodonFeature getStopCodon() {
+        return stopCodon;
+    }
+
+    // ============================================================================================================
+
+    public void setCds(GencodeGtfCDSFeature cds) {
+        this.cds = cds;
+    }
+
+    public void setStartCodon(GencodeGtfStartCodonFeature startCodon) {
+        this.startCodon = startCodon;
+    }
+
+    public void setStopCodon(GencodeGtfStopCodonFeature stopCodon) {
+        this.stopCodon = stopCodon;
+    }
+
+
+    @Override
+    protected List<GencodeGtfFeature> getAllFeatures() {
+        ArrayList<GencodeGtfFeature> list = new ArrayList<>();
+        list.add(this);
+
+        if ( cds != null ) { list.add(cds) ; }
+        if ( startCodon != null ) { list.add(startCodon) ; }
+        if ( stopCodon != null ) { list.add(stopCodon) ; }
+
+        return list;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (!(other instanceof GencodeGtfExonFeature)) {
+            return false;
+        }
+
+        GencodeGtfExonFeature otherExon = (GencodeGtfExonFeature) other;
+
+        if ( !super.equals(otherExon) ) {
+            return false;
+        }
+
+        if (!(((cds == null) && (otherExon.cds == null)) || ((cds != null) && cds.equals(otherExon.cds))) ||
```

Comment:
Use `Objects.equals()`, which will do the null checks for you (applies throughout this PR)